### ISSUE TITLE
Lack of documentation for methods in cgraph package

### ISSUE THREAD

**Issue by constraintAutomaton** [goccy/go-graphviz#74]
There are some methods in the `cgraph` package that doesn't have documentations. Their names and parameters don't necessarily make the functioning of those entities obvious, for example the method [`func (g *Graph) Degree(n *Node, in, out int) int`](https://pkg.go.dev/github.com/goccy/go-graphviz@v0.1.1/cgraph#Graph.Degree), I don't know what the `in` and `out` parameters means, for me the degree of a vertex, is simply the number of edges connected to it, so it should just have the `n` parameter. I have to look at the source code, find the `C` function underline [`agdegree`](https://github.com/goccy/go-graphviz/blob/1f0a44108c1d533c39ac8cc2b4a7b08a6fb43858/internal/ccall/cgraph/graph.c#L232) and look at the documentation of `graphViz C` implementation to figure out what that function does and the role of those parameters.

```
The function agdegree(g, n, in, out) gives the degree of a node in (sub)graph g, where in and
out select the edge sets.
• agdegree(g,n,TRUE,FALSE) returns the in-degree.
• agdegree(g,n,FALSE,TRUE) returns the out-degree.
• agdegree(g,n,TRUE,TRUE) returns their sum.
```
And some confusion remains because it seems like the parameters `in` and `out` should be boolean, at least in the `Go` implementation.

So I think that more documentation would be beneficial.

**Comment by denk0403** [comment]
While I certainly agree that the documentation on the Go bindings could be improved, the example above just feels like a case of a missed opportunity to expose a higher-level API. For instance, `cgraph` should likely expose functions called `Indegree()`, `Outdegree()`, and `TotalDegree()`. Perhaps I'll make a PR for that as well...

**Comment by goccy** [comment]
Merged #75
